pkg/ottl/ottlfuncs/func_duration_to_hours.go

```diff
@@ -0,0 +1,43 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package ottlfuncs // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/ottlfuncs"
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+)
+
+type DurationToHoursArguments[K any] struct {
+	Duration ottl.StringGetter[K] `ottlarg:"0"`
```

Comment:
These functions should work on the `time.Duration` directly.  We don't have a `ottl.DurationGetter` at the moment, though, so we'll need to add one as part of this PR.  The implementation will be very similar to `StringGetter` or other other typed Getters.

Reply:
Updated this! Sorry for misunderstanding how to construct these converters. New getters have been added.